### ISSUE TITLE
SIMKL Support

### ISSUE THREAD

**Issue by Metigoth** [jurialmunkey/plugin.video.themoviedb.helper#718]
Add support for SIMKL lists.
It supports
TV, TV Anime, Movies, Movie Anime
API documentation
https://simkl.docs.apiary.io/#

**Comment by jurialmunkey** [comment]
Interesting. What is the advantage of this over Trakt?

**Comment by Metigoth** [comment]
Just discovered it myself. It seems more responsive.
It supports anime also. It already links to multiple sources for more on TV shows and movies.
My next episodes displays better in it

**Comment by jesuslop** [comment]
> Interesting. What is the advantage of this over Trakt?

Some online oriented addons lack library integration and/or Trakt support (and use opaque plugin paths so no TMDBh player is workable). But not all is lost. For content from that sources even a database-ID lacking video can be correctly scrobbled in Simkl because as last resort something amounting to the title is sent in the scrobble request. Simkl boasts having an algorithm able to identify the ID from the title with 90%+ accuracy. I tried this to good avail. Trakt doesn't do that.

Now the problem is how to mark content in Simkl as planed to see in any platform, and then to consult it from the confort of the couch in Kodi. Surely one can do that in many ways, for instance with Trakt. The extra edge is that a simkl-only flow is more natural.

**Comment by jurialmunkey** [comment]
> > Interesting. What is the advantage of this over Trakt?
> 
> Some online oriented addons lack library integration and/or Trakt support (and use opaque plugin paths so no TMDBh player is workable). But not all is lost. For content from that sources even a database-ID lacking video can be correctly scrobbled in Simkl because as last resort something amounting to the title is sent in the scrobble request. Simkl boasts having an algorithm able to identify the ID from the title with 90%+ accuracy. I tried this to good avail. Trakt doesn't do that.
> 
> Now the problem is how to mark content in Simkl as planed to see in any platform, and then to consult it from the confort of the couch in Kodi. Surely one can do that in many ways, for instance with Trakt. The extra edge is that a simkl-only flow is more natural.

TMDbHelper doesn't handle scrobbling though because it doesn't actually play anything itself. For anything using setResolvedURL, the Trakt Kodi addon handles scrobbling. And for add-ons that don't support setResolvedURL, they often have their own scrobbling.

Point is that TMDbHelper hands off playing duties to another plugin, so it isn't suited to managing scrobbling. It can only really handle manual actions (e.g. mark as watched/unwatched from Trakt options in context menu).

**Comment by Metigoth** [comment]
There is a simkl scrobbling addon for Kodi already.

**Comment by jesuslop** [comment]
> There is a simkl scrobbling addon for Kodi already.

Exactly. The Simkl addon only scrobbles, it doesn't show our lists of content. Is a service without UI. 

That's why we ask you to provide Simkl lists in TMDBh, as you did for Trakt.  This is useful in itself. Having a TMDBh player for the content is a bonus, but even without one, seeing the lists is good per se.

**Comment by Goldenfreddy0703** [comment]
Honestly, i think this would be a great idea too cause it supports importing lists for other services and with Simkl, it syncs them and i think this would be a major feature to add cause other addons such as Kaito, Venom, Fen, Kingpin, ect may not support trakt, they could be supporting something else, That's where Simkl comes in. This would be a great idea. 
![image](https://user-images.githubusercontent.com/62456796/175844459-057fc6d8-ac8c-4475-a2e2-ac444503e0e7.png)

**Comment by Goldenfreddy0703** [comment]
Hey guys, here is the API Documentation plus some extra links on how to get the user's watchlist and how to get the items by ID.
 - Website: https://simkl.com/
 - Developer Tools: https://simkl.com/settings/developer/
 - Api Documentation: https://simkl.docs.apiary.io/ 
 - Get all items in the user's watchlist: https://simkl.docs.apiary.io/#reference/sync/get-all-items/get-all-items-in-the-user's-watchlist
 - Get items by ID: https://simkl.docs.apiary.io/#reference/redirect/mark-as-watched-from-your-website/get-items-by-id

**Comment by Goldenfreddy0703** [comment]
If we do add some categories, genres, progress, ect in the future, i can get started on the icons for it if you guys like me too.

**Comment by Goldenfreddy0703** [comment]
Ok guys, before i start making a bunch of icons, what do you guys think of the logo. 
![image](https://user-images.githubusercontent.com/62456796/176984154-8380f980-7b1a-49f6-ab7a-e87e726c9c66.png)
![Simkl Logo](https://user-images.githubusercontent.com/62456796/176983792-ad5f86bf-bf68-4046-8db2-66de6d4b41e7.png)
 - It's matching the website and has a gradient overlay background with shapes. 
 - It also matches the way TMDB and Trakt Logos were displayed

**Comment by Goldenfreddy0703** [comment]
Hey, @jurialmunkey , sorry to bother you but if you were to add Simkl support, would you also want genres with this or would you just add Simkl to the accounts category in addons settings so we can see are progress and all? If you could give me a reply on that, that would be really great to hear, Thank you

**Comment by jurialmunkey** [comment]
I haven't had a chance to have a proper look yet but I can't see any API call to retrieve available genres. I can see how it is possible to filter by genre but I can't see any way to actually retrieve the list of genres that are supported.

There's quite a few other TMDbH features that I want to get in first e.g. Trakt genres; the icon pack code; MDbList library sync support; and UpNext integration amongst other things.

So I would say hold-off for now on creating genre icons because it might be a while before I get started on adding support for SIMKL properly, and I'll only really know whether genre categories are feasible or not once I start adding it.

**Comment by Goldenfreddy0703** [comment]
Ahh I understand, thank you very much.

**Comment by Goldenfreddy0703** [comment]
Heyyyy @jurialmunkey, sorry to ping you or bother you but i was looking back at this future request about Simkl Support and i noticed you have not had a proper look at this yet with there api and all so i got in reach with the @Simkl-dev and he will be glad to answer any question or concern you have about simkl. Depending on how many people use Tmdb helper, you will need a simkl api key but after that, everything in simkl should be easy to get and all. Simkl has been amazing for me and my addon and recently my team added simkl support and tmdb player support so hopefully we will have support very soon. 

Thank you and you guys are the best!!!

**Comment by Ghb3245** [comment]
Hello 👋 

Can you please add simkl support 
I think now it got more data than tmdb and Trakt sometimes I can't find an episode released in tmdb or Trakt but I found it in simkl so if we can add the missing stuff from simkl will be great 

Have a good day

**Comment by Liquidrag** [comment]
With trait’s new 100 item limit, I would love to see simkl integration especially to access add to/remove from watchlists. @goldenfreddy0703 and his team has implemented it in Otaku. I hope you have the chance to work on this. Thank you for all your skins and addons.

**Comment by rdscorreia74** [comment]
> I hope you have the chance to work on this. 

I second that. Apparently Trakt is becoming of no use...
This would be the perfect time to come up with an alternative.

> Thank you for all your skins and addons.

Yep, thanks a bunch @jurialmunkey . You're da man

**Comment by jurialmunkey** [comment]
@Liquidrag @rdscorreia74 and others here due to recent Trakt list limits. Note that TMDbHelper also supports MDbList which does not have item limits for free users (well the limit is 10000 items per list but honestly what is even the point of a list that large!).

If your only concern is list item limits then MDbList is an option for larger lists which is already available in the plugin. I also recently added support for library monitoring from MDbLists you own.

**Comment by Goldenfreddy0703** [comment]
I initially came here for the anime features of Simkl, but the list functionality sounds incredibly useful as well. When combined with TMDb Helper, it could become even more versatile. For example, if developers of anime addons with their own databases integrated player support for TMDb Helper, users could still benefit from these features even if their addons don't directly support Trakt or TMDb. I see Simkl support as a great option because, to be honest, Trakt is becoming more limited. It frequently goes down, which is frustrating, especially when you have to reauthorize it. That's why I prefer Simkl—it’s constantly adding new features for both free and VIP users.

One day, I would love to see Simkl support for TMDb Helper and other addons in the near future. I see a lot of potential in Simkl, and I’m confident they’ll continue adding even more features in the future.

**Comment by jurialmunkey** [comment]
@Goldenfreddy0703 - I don't doubt the usefulness of SIMKL and it is likely I will implement it eventually. 

It's just a very large project to implement into TMDbHelper and I can't see myself having that time for it in at least the immediate future. That's why I mentioned MDbList, because that's an option already available now for users simply looking for larger lists.

**Comment by Goldenfreddy0703** [comment]
> @Goldenfreddy0703 - I don't doubt the usefulness of SIMKL and it is likely I will implement it eventually. 
> 
> It's just a very large project to implement into TMDbHelper and I can't see myself having that time for it in at least the immediate future. That's why I mentioned MDbList, because that's an option already available now for users simply looking for larger lists.

Ahh i understand, it is a very large project indeed and I understand that it won't be implemented for a long time. Once simkl adds more futures to there api such as discovery or more menus or watchlist next up ofcourse. It would be a very great time to implement simkl in the near future. MDbList is a great option and I use it for my anime database monthly. 

Oh btw, I'm not sure if MDbList developer told you this yet but he might have, he would like you migrate to his new api where in the api, you can make batch request now instead of making individual request. Hope this helps.

https://mdblist.docs.apiary.io/#reference/0/media-info-batch/get-media-info-batch

**Comment by jurialmunkey** [comment]
> Oh btw, I'm not sure if MDbList developer told you this yet but he might have, he would like you migrate to his new api where in the api, you can make batch request now instead of making individual request. Hope this helps.

Yes I saw that. It is not useful for TMDbHelper as it never has any need to make more than one call at a time.

MDbList ratings are only provided in the service for the currently focused item, which is retrieved on demand when it gains focus. I never get a batch of ratings for a list -- so my current approach should be better for his servers anyway (only get ratings for one item when it is needed rather than hit the server for 100s of ratings in one go that might never be used).

**Comment by Liquidrag** [comment]
Thank you @jurialmunkey i have migrated my watchlists to mdblist. Will there be an option to add to those lists through context menus or integrated into your skins like the “add to trakt” options?

**Comment by jurialmunkey** [comment]
> Thank you [@jurialmunkey](https://github.com/jurialmunkey) i have migrated my watchlists to mdblist. Will there be an option to add to those lists through context menus or integrated into your skins like the “add to trakt” options?

Yes, eventually. Those features are only available in the new MDbList API, so it is dependent on me finding the time to migrate to the new API - hopefully sooner rather than later.

**Comment by Liquidrag** [comment]
> > Thank you [@jurialmunkey](https://github.com/jurialmunkey) i have migrated my watchlists to mdblist. Will there be an option to add to those lists through context menus or integrated into your skins like the “add to trakt” options?
> 
> Yes, eventually. Those features are only available in the new MDbList API, so it is dependent on me finding the time to migrate to the new API - hopefully sooner rather than later.

I appreciate all you do. I’m just glad to know it’s coming down the pipeline sometime.

**Comment by L0bsang** [comment]
> Thank you [@jurialmunkey](https://github.com/jurialmunkey) i have migrated my watchlists to mdblist. Will there be an option to add to those lists through context menus or integrated into your skins like the “add to trakt” options?

@Liquidrag I also did the same and had a question about sorting.

I see that I can use Hide Library Items = "Not in Watchlist" to only show my watchlist items but I am not able to sort them by when they were added, which is essential for my widget display. When I use "Last Watched as a sorting method it seems to sort by Rank/Score instead. Have you found a way to do this or do you sort your watchlist differently? And how to add and remove items to it, maybe I need to make a static list?

**Comment by Liquidrag** [comment]
@L0bsang i haven’t figured that out either.

**Comment by CartesianJohn** [comment]
@jurialmunkey Given everything going on at Trakt right now, simkl seems to be a popular destination for the displaced. Any further thoughts on adding it to tmdb helper (I'm guessing you're a trakt user?).

**Comment by jurialmunkey** [comment]
> Given everything going on at Trakt right now, simkl seems to be a popular destination for the displaced. Any further thoughts on adding it to tmdb helper (I'm guessing you're a trakt user?).

As mentioned, it is a massive job.

I currently have enough on my plate as is, and after some of what I've had to put up with during recent refactoring, my enthusiasm for taking on another large project is very much dampened. 

Once this refactor for v7 of TMDbHelper is done, I'm unlikely to take on any more large projects for the foreseeable future.

**Comment by CartesianJohn** [comment]
> As mentioned, it is a massive job.
> 
> I currently have enough on my plate as is, and after some of what I've had to put up with during recent refactoring, my enthusiasm for taking on another large project is very much dampened.
> 
> Once this refactor for v7 of TMDbHelper is done, I'm unlikely to take on any more large projects for the foreseeable future.

Completely understandable. Just evaluating my options to move away from Trakt so wanted to check in :)

**Comment by chill-DA** [comment]
> As mentioned, it is a massive job.
> 
> I currently have enough on my plate as is, and after some of what I've had to put up with during recent refactoring, my enthusiasm for taking on another large project is very much dampened.
> 
> Once this refactor for v7 of TMDbHelper is done, I'm unlikely to take on any more large projects for the foreseeable future.

Have you changed your mind by any chance? Simkl integration would be highly appreciated, due to pathetic changes over at Trakt, especially their v3 UX change which removed many features. People are moving away from Trakt and Simkl seems to be a better alternative.

**Comment by CartesianJohn** [comment]
> Have you changed your mind by any chance? Simkl integration would be highly appreciated, due to pathetic changes over at Trakt, especially their v3 UX change which removed many features. People are moving away from Trakt and Simkl seems to be a better alternative.

I had a bit of a look at this last year. Unfortunately the SIMKL API didn't (and still doesn't as far as I can tell) provide access to custom lists. Without that there's really not much point imho.

**Comment by chill-DA** [comment]
> I had a bit of a look at this last year. Unfortunately the SIMKL API didn't (and still doesn't as far as I can tell) provide access to custom lists. Without that there's really not much point imho.

I messaged their support and custom lists are a part of SIMKL v2 which is currently in beta stage. Once they complete the development, they will add the lists to the API. No ETA, but it will come at some point.

**Comment by CartesianJohn** [comment]
> I messaged their support and custom lists are a part of SIMKL v2 which is currently in beta stage. Once they complete the development, they will add the lists to the API. No ETA, but it will come at some point.

That's the same reply they've been giving since they added custom lists to the v2 beta well over a year ago so don't hold your breath. I'd love to get away from Trakt, but sadly SIMKL isn't ready to be the successor yet.

**Comment by derrgoo** [comment]
Simkl is very viable at present with Trakt limiting on app authorization.
